File: ws/rojas/U2/WS15_Inheritance/ZooApp/src/main/java/ec/edu/espe/zoo/zooapp/model/Mammal.java

package ec.edu.espe.zoo.zooapp.model;

import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author USUARIO
 */
public class Mammal extends Animal {
    private float maximumHairSize;
    private int mammaryGlandsNumber;

   
    
    public void breastfeed(){
        System.out.println("breastfeeding the children ");
    }

    public Mammal(float maximumHairSize, int mammaryGlandsNumber, int id, String name, Date bornOnDate, float weight, char sex, ArrayList<Vaccine> vaccines) {
        super(id, name, bornOnDate, weight, sex, vaccines);
        this.maximumHairSize = maximumHairSize;
        this.mammaryGlandsNumber = mammaryGlandsNumber;
        
    }

    @Override
    public String toString() {
        return "Mammal{" + super.toString() + "maximumHairSize=" + maximumHairSize + ", mammaryGlandsNumber=" + mammaryGlandsNumber + '}';
    }
    
    

    /**
     * @return the maximumHairSize
     */
    public float getMaximumHairSize() {
        return maximumHairSize;
    }

    /**
     * @param maximumHairSize the maximumHairSize to set
     */
    public void setMaximumHairSize(float maximumHairSize) {
        this.maximumHairSize = maximumHairSize;
    }

    /**
     * @return the mammaryGlandsNumber
     */
    public int getMammaryGlandsNumber() {
        return mammaryGlandsNumber;
    }

    /**
     * @param mammaryGlandsNumber the mammaryGlandsNumber to set
     */
    public void setMammaryGlandsNumber(int mammaryGlandsNumber) {
        this.mammaryGlandsNumber = mammaryGlandsNumber;
    }

    public void provideVaccination(Vaccine vaccine) {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }
    
    
    
}

File: hw/quiroga/U2/Hw18Exceptions/Inheritance_JS/javascript/ec/edu/espe/zoo/model/Mammal.mjs
import { Animal } from "./Animal.mjs";

export class Mammal extends Animal {
    #maximumHairSize;
    #mammaryGlandsNumber;

    constructor(maximumHairSize, mammaryGlandsNumber, id, name, bornOnDate, weight, sex, vaccines) {
        super(id, name, bornOnDate, weight, sex, vaccines);
        this.#maximumHairSize = maximumHairSize;
        this.#mammaryGlandsNumber = mammaryGlandsNumber;
    }

    toString() {
        return `Mammal {${super.toString()}, maximumHairSize: ${this.#maximumHairSize}, mammaryGlandsNumber: ${this.#mammaryGlandsNumber}}`;
    }

    breastFeed() {
        console.log("Breastfeeding the children");
    }

    getMaximumHairSize() {
        return this.#maximumHairSize;
    }

    setMaximumHairSize(maximumHairSize) {
        this.#maximumHairSize = maximumHairSize;
    }

    getMammaryGlandsNumber() {
        return this.#mammaryGlandsNumber;
    }

    setMammaryGlandsNumber(mammaryGlandsNumber) {
        this.#mammaryGlandsNumber = mammaryGlandsNumber;
    }
}
